docs/challenges/language-support/ocaml.mdx

```diff
@@ -0,0 +1,87 @@
+---
+title: "OCaml"
+description: "Learn how to solve CodeCrafters challenges in OCaml"
+---
+
+<Note>
+  **New to OCaml?** We recommend checking out [A Tour of
+  OCaml](https://ocaml.org/docs/tour-of-ocaml) to get up to speed.
+</Note>
+
+# Local development
+
+To run the code for your challenge locally, you'll need [opam](https://opam.ocaml.org/) and [dune](https://dune.build/install) installed. Our test runners use Dune v3.16 (as of September 2024).
+
+The script for your challenge (`./your_program.sh`) will run your code using these tools. It'll look something like this:
+
+```sh
+#!/bin/sh
+
+dune build --build-dir /tmp/redis
+exec /tmp/redis/default/main.exe "$@"
+```
+
+# File structure
+
+- The files for your solution are placed in the `src/` directory.
+- `src/main.ml` is the entry point for your OCaml program.
+
+# Adding more files
+
+You can organize your code by adding more `.ml` files in the `src/` directory. The test runner will include them when building and running your code.
+
+For example, if you add a file at `src/foo.ml`, you can reference it in `main.ml` like this:
+
+<CodeGroup>
+
+```ocaml src/foo.ml
+let bar () = Printf.printf "Hello World!\n"
+```
+
+```ocaml src/main.ml
+let () = Foo.bar () (* Calls the bar function from foo.ml *)
+```
+
+</CodeGroup>
+
+# Adding dependencies
+
+You can add dependencies by editing the `.opam` file and the `dune` file.
```

Comment:
- Edit `dune-project`
- Edit `dune`
- Run `opam exec -- dune build` to regenerate the `.opam` file
- Run `opam install . --deps-only` to install the newly added dependency. 
- Commit changes to all three files (`dune`, `dune-project` and `*.opam`).